pkg/signal/trap_linux_test.go

```diff
@@ -0,0 +1,66 @@
+// +build linux
+
+package signal
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTrap(t *testing.T) {
+	sigmap := map[string]os.Signal{
+		"TERM": syscall.SIGTERM,
+		"QUIT": syscall.SIGQUIT,
+		"INT":  os.Interrupt,
+	}
+	for k, v := range sigmap {
+		wd, _ := os.Getwd()
+		testHelperCode := wd + "/testfiles/main.go"
+		cmd := exec.Command("go", "build", "-o", "main", testHelperCode)
```

Comment:
I think this binary can be created once outside of the loop. It doesn't need to be rebuilt each time.

Maybe `s/testHelperCode/testHelperPath/` since the variable is a file path, not code.

I think the binary should also go into a temporary directory, instead of the current working director. The temporary directory can be cleaned up with a `defer os.RemoveAll()` as well.

I'm not really sure if `os.Getwd()` is always going to be the package directory, but if it works for now, I guess it's fine.